sdk/eventhub/azure-eventhubs/api_review.py

```diff
@@ -0,0 +1,235 @@
+
+class EventHubProducerClient(ClientBaseAsync):
+
+    eh_name  # type: str
+
+    def __init__(
+            self,
+            host: str,
+            event_hub_path: str,
+            credential: TokenCredential,
+            **kwargs
+    ) -> None:
+
+    async def __aenter__(self):
+
+    async def __aexit__(self, *args):
+
+    @classmethod
+    def from_connection_string(
+        cls, conn_str: str,
+        *,
+        event_hub_path: str = None,
+        logging_enable: bool = False,
+        http_proxy: dict = None,
+        auth_timeout: float = 60,
+        user_agent: str = None,
+        retry_total: int = 3,
+        transport_type=None) -> EventHubProducerClient:
+
+    async def send(
+        self,
+        event_data: Union[EventData, EventDataBatch, Iterable[EventData]],
+        *,
+        partition_key: Union[str, bytes] = None,
+        partition_id: str = None,
+        timeout: float = None) -> None:
+
+    async def create_batch(self, events: Iterable[EventData] = None, max_size: int = None) -> EventDataBatch:
+
+    async def get_properties(self) -> Dict[str, Any]:
+
+    async def get_partition_ids(self) -> List[str]:
+
+    async def get_partition_properties(self, partition: str) -> Dict[str, str]:
```

Comment:
We should either consistently use `partition_id` or `partition`.

Reply:
This also applies to `consumer_group` vs `consumer_group_name`